speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/unitTestUtilitiesTests/ComparerUtilsTests.cpp

```diff
@@ -0,0 +1,153 @@
+    ComparerUtils::CompareCars(car1, car2, false);                                                                                  // Test pointer inequality
+    ComparerUtils::CompareSituations(situation1, situation2, true);                                                                 // Test value equality
+    ComparerUtils::CompareSituations(situation1, situation2, false);                                                                // Test pointer inequality
+    ComparerUtils::CompareSegments(segments1, segments2, segmentCountTest, true);                                                   // Test value equality
+    ComparerUtils::CompareSegments(segments1, segments2, segmentCountTest, false);                                                  // Test pointer inequality
+
+    delete[] segments1;
+    delete[] segments2;
+}
+
+/// @brief Tests whether comparisons work correctly between randomly-generated structs
+TEST(ComparerUtilsTests, GeneratedCompare)
+{
+    TestSegments segments1 = GeneratorUtils::GenerateSegments();
+    tTrackSeg* segments2 = new tTrackSeg[segments1.NextSegmentsCount];
+    segments2[0] = {};
+    for (int i = 1; i < segments1.NextSegmentsCount; i++)
+    {
+        segments2[i] = {};
+        segments2[i - 1].next = &segments2[i];
+    }
+    segments2[segments1.NextSegmentsCount / 2].cos = segments1.NextSegments[segments1.NextSegmentsCount / 2].cos + 1;  // Guarantee difference
+    tCarElt car1 = {};
+    tCarElt car2 = GeneratorUtils::GenerateCar(segments1);
+    car1.index = car2.index + 1;  // Guarantee difference
+    tSituation situation1 = {};
+    tSituation situation2 = GeneratorUtils::GenerateSituation();
+    situation2.raceInfo.features = situation1.raceInfo.features + 1;  // Guaranteee difference
+
+    INVERT_COMP_TEST(ComparerUtils::CompareSegments(segments1.NextSegments, segments2, segments1.NextSegmentsCount, true))  // Test value inequality
+    INVERT_COMP_TEST(ComparerUtils::CompareCars(car1, car2, true))                                                          // Test value inequality
+    INVERT_COMP_TEST(ComparerUtils::CompareSituations(situation1, situation2, true))                                        // Test value inequality
+
+    delete[] segments2;
+    GeneratorUtils::DestroySituation(situation2);
+    GeneratorUtils::DestroyCar(car2);
+    GeneratorUtils::DestroySegments(segments1);
+}
+
+/// @brief Tests whether comparisons work even if only one element differs
+TEST(ComparerUtilsTests, AlmostEqual)
+{
+    TestSegments segments1 = GeneratorUtils::GenerateSegments();
+    tCarElt car1 = GeneratorUtils::GenerateCar(segments1);
+    tCarElt car2 = car1;
+    car2.pitcmd.stopType++;
+    tSituation situation1 = GeneratorUtils::GenerateSituation();
+    tSituation situation2 = situation1;
+    situation2.deltaTime++;
+    constexpr int segmentCountTest = 10;
+    tTrackSeg* segments2 = new tTrackSeg[segmentCountTest];
+    tTrackSeg* segments3 = new tTrackSeg[segmentCountTest];
```

Comment:
This can be changed to use a define for segmentCountTest and use that define to create stack allocated arrays